### PR TITLE
Initial set off iOS e2e tests

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Adds and initial set of end-to-end tests for iOS.  

## Design

Follow the existing approach in place for Android.  A number of follow on tasks have been raised, as noted in `TODO`s.

## Changeset

- Initial set of iOS e2e test scenarios
- Test fixture button layout updated to start at the top and flow down (avoiding problems with different screen dimensions, as Appium considers the origin to be at the top-left).
- Test fixture button text updated to reflect what each now does.

## Testing

Covered by CI and also run successfully locally several times during development.

### CODE REVIEW

File: src/BugsnagUnity.mm [hunk 1]
```diff
@@ -109,6 +109,8 @@ void bugsnag_setNotifyUrl(const void *configuration, char *notifyURL) {
     return;
   NSString *ns_notifyURL = [NSString stringWithUTF8String: notifyURL];
   ((__bridge BugsnagConfiguration *)configuration).endpoints.notify = ns_notifyURL;
+  // Workaround for endpoint stale-cache issue: Force-trigger re-processing by reassinging endpoint
```
Comment (fractalwrench) [inline src/BugsnagUnity.mm:112]: Has a bug been raised for this on the Cocoa side? It'd be good to link to it if so, or create one if not.

File: test/mobile/features/ios/ios_sessions.feature [hunk 2]
```diff
@@ -0,0 +1,55 @@
+Feature: iOS smoke tests for sessions
+
+    # TODO: Failing steps commented out pending full implementation of PLAT-6372
```
Comment (fractalwrench) [inline test/mobile/features/ios/ios_sessions.feature:3]: This looks like it might be obsolete

**Review by fractalwrench — APPROVED** [review]
LGTM based on the CI results 🎉

File: test/mobile/features/ios/ios_sessions.feature (hunk 2 above)
Comment (twometresteve) [reply test/mobile/features/ios/ios_sessions.feature:3]: Ah yes.

Comment (twometresteve) [reply test/mobile/features/ios/ios_sessions.feature:3]: .... removed on my next branch to avoid another CI run.

**Pull request merged**